core/src/main/scala/org/apache/spark/api/r/SerDe.scala

```diff
@@ -46,9 +46,18 @@ private[spark] object SerDe {
     dis.readByte().toChar
   }
 
-  def readObject(dis: DataInputStream): Object = {
+  def readObject(dis: DataInputStream, typeName: String = ""): Object = {
```

Comment:
Currently, `SerDe` is clean and doesn't include any import from sql and I think it shouldn't too because it is in core? So I just use string here.